go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/argoproj/argo-cd/v2
 
-go 1.22.0
+go 1.23.0
```

Comment:
we will have to port over this change too https://github.com/argoproj/argo-cd/pull/21748/files#r1968426728